library/core/src/char/methods.rs

```diff
@@ -1544,9 +1544,18 @@ impl char {
     #[rustc_const_stable(feature = "const_ascii_ctype_on_intrinsics", since = "1.47.0")]
     #[inline]
     pub const fn is_ascii_whitespace(&self) -> bool {
-        match *self {
-            '\t' | '\n' | '\x0C' | '\r' | ' ' => true,
-            _ => false,
+        #[cfg(target_pointer_width = "64")]
```

Comment:
Seems a bit unfortunate of a gate as target pointer width doesn't always correlate with whether 64-bit registers are available, such as on the x32 ABI and WASM32.

Reply:
Is there a better way to detect 64-bit platforms (besides enumerating them)?

An alternative here would be to convert `*self` to `u32`, then subtract 1. It's an extra instruction, but the mask check can be done on an `u32` afterwards, so this would leave 16-bit platforms (AVR?).